Portal/src/Datahub.Infrastructure/Services/UsersStatusService.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Datahub.Application.Configuration;
+using Datahub.Application.Services;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Datahub.Infrastructure.Services
+{
+    public partial class UsersStatusService : IUsersStatusService
+    {
+        private readonly ILogger<UsersStatusService> _logger;
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly DatahubPortalConfiguration _datahubPortalConfiguration;
+        
+        public UsersStatusService(ILogger<UsersStatusService> logger, IHttpClientFactory httpClientFactory,
+            DatahubPortalConfiguration datahubPortalConfiguration)
+        {
+            _logger = logger;
+            _httpClientFactory = httpClientFactory;
+            _datahubPortalConfiguration = datahubPortalConfiguration;
+        }
+        
+        public async Task<Dictionary<string, List<string>>> GetUsersStatus()
+        {
+            var url = _datahubPortalConfiguration.DatahubGraphUsersStatusFunctionUrl;
+            
+            var numberOfRetries = 0;
+            const int maxNumberOfRetries = 5;
+            string resultString;
+            
+            do
+            {
+                using var client = _httpClientFactory.CreateClient();
+                var result = await client.GetAsync(url);
+                resultString = await result.Content.ReadAsStringAsync();
+            
+            } while (string.IsNullOrWhiteSpace(resultString) && numberOfRetries++ < maxNumberOfRetries);
```

Comment:
This retry is probably because you get a throttling response from Azure (409), you can use a Retry Policy as we do with the Azure Management API in the Costing and Storage stats collection. I can show you if that is the case of the problem.

Reply:
In most recent commit, added a proper retry policy, please let me know if it was implemented properly